### PR TITLE
Data API Template App (pt2)

### PR DESCRIPTION
**Pull request by nlarew**
## Changes

- Switches from constructed base URLs to sourcing them from the metadata file
- Adds a manifest entry for the Data API template
- Renames metadata files for all web templates from `realm.json` to `atlasConfig.json`
- Adds a GitHub action to automate testing
- Adds `create-metadata-file.sh` to generate `realm.json`/`atlasConfig.json` files
- Updates the README
- Switches from `create-react-app` to `vite`
- Switches from `jest` to `vitest`
- Switches from `.js` to `.jsx` file extensions

## Jira

- [(DOCSP-28716) Integrate API Base URLs from realm.json](https://jira.mongodb.org/browse/DOCSP-28716)
- [(DOCSP-28038) Create a manifest entry for the Data API template](https://jira.mongodb.org/browse/DOCSP-28038)
- [(DOCSP-28725) Switch from CRA to Vite](https://jira.mongodb.org/browse/DOCSP-28725)

## How To Test

This PR already ran the automated testing GHA. You can use [act](https://github.com/nektos/act) to run the GHA locally from the repo root (you will need to provide an Atlas API Key pair):

```sh
ATLAS_PUBLIC_API_KEY="abcdefgh"
ATLAS_PRIVATE_API_KEY="11111111-0ff0-1337-h4c2-f00b470ec112"
act -j web-js-integration-test -s ATLAS_PUBLIC_API_KEY -s ATLAS_PRIVATE_API_KEY --container-architecture linux/amd64
```

Or you can manually set up and run the tests

1. Navigate to the project root
   
   ```
   cd other/web-js
   ```

2. Create backend Apps
   
   ```
   # Backend for prod-mql and prod-graphql
   cd backend
   npx mongodb-realm-cli push
   cd ..
   # Backend for prod-data-api   
   cd backend-data-api
   npx mongodb-realm-cli push
   cd ..
   ```

3. Create metadata files

   In the repo root, find `create-metadata-file.sh` and call it to create metadata files for the dev client & the generated projects. You may need to run `chmod +x create-metadata-file.sh` first.
   
   ```
   ./create-metadata-file.sh other/web-js/backend-data-api other/web-js/client/src/atlasConfig.json
   ./create-metadata-file.sh other/web-js/backend-data-api other/web-js/generated/prod-data-api/src/atlasConfig.json
   ./create-metadata-file.sh other/web-js/backend other/web-js/generated/prod-graphql/src/atlasConfig.json
   ./create-metadata-file.sh other/web-js/backend other/web-js/generated/prod-mql/src/atlasConfig.json
   ```

4. Run the combined tests - these run on the development code
   
   ```
   cd client
   npm install
   npm run test:mql
   npm run test:graphql
   npm run test:data-api
   cd ..
   ```

5. Run the artifact tests - these run on the generated code. Do this for each artifact project.
   
   ```
   cd generated/prod-data-api
   # cd generated/prod-graphql
   # cd generated/prod-mql
   npm install
   npm run test
   ```

### CODE REVIEW

File: .github/workflows/test-web-js.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            test_backend_dir: backend-data-api-test
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+      - name: CLI Authentication
+        run: |
+          npx mongodb-realm-cli login --api-key=${{ secrets.ATLAS_PUBLIC_API_KEY }} --private-api-key=${{ secrets.ATLAS_PRIVATE_API_KEY }}
+          npx mongodb-realm-cli whoami
```
Comment (mongodben) [inline .github/workflows/test-web-js.yml:35]: can this be removed?

**Comment by mongodben** [comment]
couple overarching comments about maintainability of this approach w 3 template apps in 1:

- there are a bunch of scripts in here to manage the 3 apps in 1. in a README/CONTRIBUTING file, can you provide a high level overview of how to make updates to the web template apps? 

- could also include the testing information that you include in the PR description in that file? 

i think future folks who touch this project (or your future self) will thank your present self for a bit of additional direction here.

**Comment by nlarew** [comment]
Added a Contributing section to the client README with new instructions about using the Bluehawk states + testing

**Review by mongodben — APPROVED** [review]
lgtm, nice work here!

**Pull request merged**